app/Services/Patreon/PatreonApi.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace App\Services\Patreon;
+
+use App\Services\Patreon\Resources\Campaign;
+use App\Services\Patreon\Resources\Pledge;
+use App\Services\Patreon\Resources\ResourceCollection;
+use App\Services\Patreon\Resources\Reward;
+use App\Services\Patreon\Resources\User;
+use GuzzleHttp\Client;
+use Illuminate\Support\Collection;
+
+class PatreonApi
+{
+    /** @var \GuzzleHttp\Client */
+    protected $client;
+
+    public function __construct(string $accessToken)
```

Comment:
Accept a `Client` as a constructor argument. Move the setup of that client to a serviceprovider.